signer/src/api/new_block.rs

```diff
@@ -32,39 +26,119 @@ pub type StacksNodePayload = Result<Json<serde_json::Value>, JsonRejection>;
+        let res = match RegistryEvent::try_from_value(ev.value, network) {
+            Ok(RegistryEvent::CompletedDeposit(event)) => {
+                api.db.write_completed_deposit_event(&event).await
+            }
+            Ok(RegistryEvent::WithdrawalAccept(event)) => {
+                api.db.write_withdrawal_accept_event(&event).await
+            }
+            Ok(RegistryEvent::WithdrawalCreate(event)) => {
+                api.db.write_withdrawal_create_event(&event).await
+            }
+            Ok(RegistryEvent::WithdrawalReject(event)) => {
+                api.db.write_withdrawal_reject_event(&event).await
+            }
+            Err(err) => {
+                tracing::error!("Got an error when transforming the event ClarityValue: {err}");
+                return StatusCode::OK;
+            }
+        };
+        // If we got an error writing to the database, this might be an
+        // issue that will resolve itself if we try again in a few moments.
+        // So we return a non success status code so that the node retries
+        // in a second.
+        if let Err(err) = res {
+            tracing::error!("Got an error when writing event to database: {err}");
+            return StatusCode::INTERNAL_SERVER_ERROR;
+        }
+    }
+
     StatusCode::OK
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    use bitcoin::OutPoint;
+    use test_case::test_case;
+
+    use crate::config::Settings;
+    use crate::storage::in_memory::Store;
+
+    /// These were generated from a stacks node after running the
+    /// "complete-deposit standard recipient", "accept-withdrawal",
+    /// "create-withdrawal", and "reject-withdrawal" variants,
+    /// respectively, of the `complete_deposit_wrapper_tx_accepted`
+    /// integration test.
+    const COMPLETED_DEPOSIT_WEBHOOK: &str =
+        include_str!("../../tests/fixtures/complete-deposit-event.json");
+
+    const WITHDRAWAL_ACCEPT_WEBHOOK: &str =
+        include_str!("../../tests/fixtures/withdrawal-accept-event.json");
+
+    const WITHDRAWAL_CREATE_WEBHOOK: &str =
+        include_str!("../../tests/fixtures/withdrawal-create-event.json");
+
+    const WITHDRAWAL_REJECT_WEBHOOK: &str =
+        include_str!("../../tests/fixtures/withdrawal-reject-event.json");
+
+    #[test_case(COMPLETED_DEPOSIT_WEBHOOK, |db| db.completed_deposit_events.get(&OutPoint::null()).is_none(); "completed-deposit")]
+    #[test_case(WITHDRAWAL_CREATE_WEBHOOK, |db| db.withdrawal_create_events.get(&1).is_none(); "withdrawal-create")]
+    #[test_case(WITHDRAWAL_ACCEPT_WEBHOOK, |db| db.withdrawal_accept_events.get(&1).is_none(); "withdrawal-accept")]
+    #[test_case(WITHDRAWAL_REJECT_WEBHOOK, |db| db.withdrawal_reject_events.get(&2).is_none(); "withdrawal-reject")]
+    #[tokio::test]
+    async fn test_events<F>(body_str: &str, func: F)
+    where
+        F: Fn(tokio::sync::MutexGuard<'_, Store>) -> bool,
+    {
+        let config_path = Some("./src/config/default");
```

Comment:
Maybe we should actually make this a const in `config` (i.e. move the const from `config::tests` that i added to the `config` module with a `#[cfg(test)]` on it), since it seems we'll be using this in multiple places.

Reply:
Good idea, will do.

Reply:
I ended up moving it to the testing module, in case we need it for our integration tests.